ext/session/session.c

```diff
@@ -2711,6 +2723,22 @@ static const zend_function_entry session_functions[] = {
 
 /* {{{ SessionHandlerInterface functions[]
 */
+static const zend_function_entry php_session_save_handler_iface_functions[] = {
+	PHP_ABSTRACT_ME(SessionSaveHandlerInterface, open, arginfo_session_class_open)
```

Comment:
> Bloated code is intentional.

Seriously, when designing a new interface, get this out of your mind.

If you want to design an API, do it properly or don't do it at all. This API, once (and if) merged, will stick around for AGES, so it better be good.

Details like:

 * close
 * gc
 * createId
 * validateId
 * updateTimestamp

are not necessary, and just move time-based coupling from the domain of the extension, making things much easier to break.